<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Workflows for analyzing single-cell RNA-seq data with R/Bioconductor</title>
  <meta name="description" content="Workflows for analyzing single-cell RNA-seq data with R/Bioconductor">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="Workflows for analyzing single-cell RNA-seq data with R/Bioconductor" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Workflows for analyzing single-cell RNA-seq data with R/Bioconductor" />
  
  
  



<meta name="date" content="2018-10-25">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="visualizing-data-in-low-dimensional-space.html">
<link rel="next" href="concluding-remarks.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Low Level scRNA-seq Workflow</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Intro</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#workflow-version-information"><i class="fa fa-check"></i><b>1.1</b> Workflow version information</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#motivation"><i class="fa fa-check"></i><b>1.2</b> Motivation</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#scrna-seq-data-analysis-with-bioconductor"><i class="fa fa-check"></i><b>1.3</b> scRNA-seq data analysis with Bioconductor</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#obtaining-a-count-matrix"><i class="fa fa-check"></i><b>1.4</b> Obtaining a count matrix</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#author-information"><i class="fa fa-check"></i><b>1.5</b> Author information</a><ul>
<li class="chapter" data-level="1.5.1" data-path="index.html"><a href="index.html#author-contributions"><i class="fa fa-check"></i><b>1.5.1</b> Author contributions</a></li>
<li class="chapter" data-level="1.5.2" data-path="index.html"><a href="index.html#competing-interests"><i class="fa fa-check"></i><b>1.5.2</b> Competing interests</a></li>
<li class="chapter" data-level="1.5.3" data-path="index.html"><a href="index.html#grant-information"><i class="fa fa-check"></i><b>1.5.3</b> Grant information</a></li>
<li class="chapter" data-level="1.5.4" data-path="index.html"><a href="index.html#acknowledgements"><i class="fa fa-check"></i><b>1.5.4</b> Acknowledgements</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="overview.html"><a href="overview.html"><i class="fa fa-check"></i><b>2</b> Overview</a></li>
<li class="chapter" data-level="3" data-path="setting-up-the-data.html"><a href="setting-up-the-data.html"><i class="fa fa-check"></i><b>3</b> Setting up the data</a><ul>
<li class="chapter" data-level="3.1" data-path="setting-up-the-data.html"><a href="setting-up-the-data.html#loading-in-the-count-matrix"><i class="fa fa-check"></i><b>3.1</b> Loading in the count matrix</a></li>
<li class="chapter" data-level="3.2" data-path="setting-up-the-data.html"><a href="setting-up-the-data.html#incorporating-cell-based-annotation"><i class="fa fa-check"></i><b>3.2</b> Incorporating cell-based annotation</a></li>
<li class="chapter" data-level="3.3" data-path="setting-up-the-data.html"><a href="setting-up-the-data.html#incorporating-gene-based-annotation"><i class="fa fa-check"></i><b>3.3</b> Incorporating gene-based annotation</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="quality-control-on-the-cells.html"><a href="quality-control-on-the-cells.html"><i class="fa fa-check"></i><b>4</b> Quality control on the cells</a><ul>
<li class="chapter" data-level="4.1" data-path="quality-control-on-the-cells.html"><a href="quality-control-on-the-cells.html#defining-the-quality-control-metrics"><i class="fa fa-check"></i><b>4.1</b> Defining the quality control metrics</a></li>
<li class="chapter" data-level="4.2" data-path="quality-control-on-the-cells.html"><a href="quality-control-on-the-cells.html#identifying-outliers-for-each-metric"><i class="fa fa-check"></i><b>4.2</b> Identifying outliers for each metric</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="classification-of-cell-cycle-phase.html"><a href="classification-of-cell-cycle-phase.html"><i class="fa fa-check"></i><b>5</b> Classification of cell cycle phase</a></li>
<li class="chapter" data-level="6" data-path="examining-gene-level-expression-metrics.html"><a href="examining-gene-level-expression-metrics.html"><i class="fa fa-check"></i><b>6</b> Examining gene-level expression metrics</a><ul>
<li class="chapter" data-level="6.1" data-path="examining-gene-level-expression-metrics.html"><a href="examining-gene-level-expression-metrics.html#inspecting-the-most-highly-expressed-genes"><i class="fa fa-check"></i><b>6.1</b> Inspecting the most highly expressed genes</a></li>
<li class="chapter" data-level="6.2" data-path="examining-gene-level-expression-metrics.html"><a href="examining-gene-level-expression-metrics.html#filtering-out-low-abundance-genes"><i class="fa fa-check"></i><b>6.2</b> Filtering out low-abundance genes</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="normalization-of-cell-specific-biases.html"><a href="normalization-of-cell-specific-biases.html"><i class="fa fa-check"></i><b>7</b> Normalization of cell-specific biases</a><ul>
<li class="chapter" data-level="7.1" data-path="normalization-of-cell-specific-biases.html"><a href="normalization-of-cell-specific-biases.html#using-the-deconvolution-method-to-deal-with-zero-counts"><i class="fa fa-check"></i><b>7.1</b> Using the deconvolution method to deal with zero counts</a></li>
<li class="chapter" data-level="7.2" data-path="normalization-of-cell-specific-biases.html"><a href="normalization-of-cell-specific-biases.html#computing-separate-size-factors-for-spike-in-transcripts"><i class="fa fa-check"></i><b>7.2</b> Computing separate size factors for spike-in transcripts</a></li>
<li class="chapter" data-level="7.3" data-path="normalization-of-cell-specific-biases.html"><a href="normalization-of-cell-specific-biases.html#applying-the-size-factors-to-normalize-gene-expression"><i class="fa fa-check"></i><b>7.3</b> Applying the size factors to normalize gene expression</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="modelling-the-technical-noise-in-gene-expression.html"><a href="modelling-the-technical-noise-in-gene-expression.html"><i class="fa fa-check"></i><b>8</b> Modelling the technical noise in gene expression</a></li>
<li class="chapter" data-level="9" data-path="removing-the-batch-effect.html"><a href="removing-the-batch-effect.html"><i class="fa fa-check"></i><b>9</b> Removing the batch effect</a></li>
<li class="chapter" data-level="10" data-path="denoising-expression-values-using-pca.html"><a href="denoising-expression-values-using-pca.html"><i class="fa fa-check"></i><b>10</b> Denoising expression values using PCA</a></li>
<li class="chapter" data-level="11" data-path="visualizing-data-in-low-dimensional-space.html"><a href="visualizing-data-in-low-dimensional-space.html"><i class="fa fa-check"></i><b>11</b> Visualizing data in low-dimensional space</a><ul>
<li class="chapter" data-level="11.1" data-path="visualizing-data-in-low-dimensional-space.html"><a href="visualizing-data-in-low-dimensional-space.html#with-pca"><i class="fa fa-check"></i><b>11.1</b> With PCA</a></li>
<li class="chapter" data-level="11.2" data-path="visualizing-data-in-low-dimensional-space.html"><a href="visualizing-data-in-low-dimensional-space.html#with-t-sne"><i class="fa fa-check"></i><b>11.2</b> With <em>t</em>-SNE</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="clustering-cells-into-putative-subpopulations.html"><a href="clustering-cells-into-putative-subpopulations.html"><i class="fa fa-check"></i><b>12</b> Clustering cells into putative subpopulations</a><ul>
<li class="chapter" data-level="12.1" data-path="clustering-cells-into-putative-subpopulations.html"><a href="clustering-cells-into-putative-subpopulations.html#defining-cell-clusters-from-expression-data"><i class="fa fa-check"></i><b>12.1</b> Defining cell clusters from expression data</a></li>
<li class="chapter" data-level="12.2" data-path="clustering-cells-into-putative-subpopulations.html"><a href="clustering-cells-into-putative-subpopulations.html#detecting-marker-genes-between-clusters"><i class="fa fa-check"></i><b>12.2</b> Detecting marker genes between clusters</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="concluding-remarks.html"><a href="concluding-remarks.html"><i class="fa fa-check"></i><b>13</b> Concluding remarks</a></li>
<li class="chapter" data-level="14" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>14</b> References</a></li>
<li class="chapter" data-level="" data-path="references-1.html"><a href="references-1.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/AJABADI/" target="blank">See on GitHub</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Workflows for analyzing single-cell RNA-seq data with R/Bioconductor</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="clustering-cells-into-putative-subpopulations" class="section level1">
<h1><span class="header-section-number">Chapter 12</span> Clustering cells into putative subpopulations</h1>
<div id="defining-cell-clusters-from-expression-data" class="section level2">
<h2><span class="header-section-number">12.1</span> Defining cell clusters from expression data</h2>
<p>The denoised log-expression values are used to cluster cells into putative subpopulations.
Specifically, we perform hierarchical clustering on the Euclidean distances between cells, using Ward’s criterion to minimize the total variance within each cluster.
This yields a dendrogram that groups together cells with similar expression patterns across the chosen genes.</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb66-1" data-line-number="1">pcs &lt;-<span class="st"> </span><span class="kw">reducedDim</span>(sce, <span class="st">&quot;PCA&quot;</span>)</a>
<a class="sourceLine" id="cb66-2" data-line-number="2">my.dist &lt;-<span class="st"> </span><span class="kw">dist</span>(pcs)</a>
<a class="sourceLine" id="cb66-3" data-line-number="3">my.tree &lt;-<span class="st"> </span><span class="kw">hclust</span>(my.dist, <span class="dt">method=</span><span class="st">&quot;ward.D2&quot;</span>)</a></code></pre></div>
<p>Clusters are explicitly defined by applying a dynamic tree cut <span class="citation">(Langfelder, Zhang, and Horvath <a href="#ref-langfelder2008defining">2008</a>)</span> to the dendrogram.
This exploits the shape of the branches in the dendrogram to refine the cluster definitions, and is more appropriate than <code>cutree</code> for complex dendrograms.
Greater control of the empirical clusters can be obtained by manually specifying <code>cutHeight</code> in <code>cutreeDynamic</code>.
We also set <code>minClusterSize</code> to a lower value than the default of 20, to avoid spurious aggregation of distant small clusters.</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb67-1" data-line-number="1"><span class="kw">library</span>(dynamicTreeCut)</a>
<a class="sourceLine" id="cb67-2" data-line-number="2">my.clusters &lt;-<span class="st"> </span><span class="kw">unname</span>(<span class="kw">cutreeDynamic</span>(my.tree, <span class="dt">distM=</span><span class="kw">as.matrix</span>(my.dist), </a>
<a class="sourceLine" id="cb67-3" data-line-number="3">    <span class="dt">minClusterSize=</span><span class="dv">10</span>, <span class="dt">verbose=</span><span class="dv">0</span>))</a></code></pre></div>
<p>We examine the distribution of cells in each cluster with respect to known factors.
Each cluster is comprised of cells from both batches, indicating that the clustering is not driven by a batch effect.
Differences in the composition of each cluster are observed with respect to <code>Oncogene</code>, consistent with a biological effect of oncogene induction.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb68-1" data-line-number="1"><span class="kw">table</span>(my.clusters, sce<span class="op">$</span>Plate)</a></code></pre></div>
<pre><code>##            
## my.clusters 20160113 20160325
##           1       40       39
##           2       19       17
##           3       16       14
##           4       10       14
##           5        6        8</code></pre>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb70-1" data-line-number="1"><span class="kw">table</span>(my.clusters, sce<span class="op">$</span>Oncogene)</a></code></pre></div>
<pre><code>##            
## my.clusters induced control
##           1      79       0
##           2       0      36
##           3       0      30
##           4       0      24
##           5      14       0</code></pre>
<p>We visualize the cluster assignments for all cells on the <em>t</em>-SNE plot in Figure <a href="clustering-cells-into-putative-subpopulations.html#fig:tsnecluster416b">12.1</a>.
Adjacent cells are generally assigned to the same cluster, indicating that the clustering procedure was applied correctly.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb72-1" data-line-number="1">sce<span class="op">$</span>cluster &lt;-<span class="st"> </span><span class="kw">factor</span>(my.clusters)</a>
<a class="sourceLine" id="cb72-2" data-line-number="2"><span class="kw">plotTSNE</span>(sce, <span class="dt">colour_by=</span><span class="st">&quot;cluster&quot;</span>) <span class="op">+</span><span class="st"> </span>fontsize</a></code></pre></div>
<div class="figure"><span id="fig:tsnecluster416b"></span>
<img src="simpleSingleCell_files/figure-html/tsnecluster416b-1.png" alt="_t_-SNE plot of the denoised PCs of the 416B dataset. Each point represents a cell and is coloured according to the cluster identity to which it was assigned." width="672" />
<p class="caption">
Figure 12.1: <em>t</em>-SNE plot of the denoised PCs of the 416B dataset. Each point represents a cell and is coloured according to the cluster identity to which it was assigned.
</p>
</div>
<p>We check the separatedness of the clusters using the silhouette width (Figure <a href="clustering-cells-into-putative-subpopulations.html#fig:silhouette416b">12.2</a>).
Cells with large positive silhouette widths are closer to other cells in the <em>same</em> cluster than to cells in <em>different</em> clusters.
Conversely, cells with negative widths are closer to other clusters than to other cells in the cluster to which it was assigned.
Each cluster would ideally contain many cells with large positive widths, indicating that it is well-separated from other clusters.</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb73-1" data-line-number="1"><span class="kw">library</span>(cluster)</a>
<a class="sourceLine" id="cb73-2" data-line-number="2">clust.col &lt;-<span class="st"> </span>scater<span class="op">:::</span><span class="kw">.get_palette</span>(<span class="st">&quot;tableau10medium&quot;</span>) <span class="co"># hidden scater colours</span></a>
<a class="sourceLine" id="cb73-3" data-line-number="3">sil &lt;-<span class="st"> </span><span class="kw">silhouette</span>(my.clusters, <span class="dt">dist =</span> my.dist)</a>
<a class="sourceLine" id="cb73-4" data-line-number="4">sil.cols &lt;-<span class="st"> </span>clust.col[<span class="kw">ifelse</span>(sil[,<span class="dv">3</span>] <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>, sil[,<span class="dv">1</span>], sil[,<span class="dv">2</span>])]</a>
<a class="sourceLine" id="cb73-5" data-line-number="5">sil.cols &lt;-<span class="st"> </span>sil.cols[<span class="kw">order</span>(<span class="op">-</span>sil[,<span class="dv">1</span>], sil[,<span class="dv">3</span>])]</a>
<a class="sourceLine" id="cb73-6" data-line-number="6"><span class="kw">plot</span>(sil, <span class="dt">main =</span> <span class="kw">paste</span>(<span class="kw">length</span>(<span class="kw">unique</span>(my.clusters)), <span class="st">&quot;clusters&quot;</span>), </a>
<a class="sourceLine" id="cb73-7" data-line-number="7">    <span class="dt">border=</span>sil.cols, <span class="dt">col=</span>sil.cols, <span class="dt">do.col.sort=</span><span class="ot">FALSE</span>) </a></code></pre></div>
<div class="figure"><span id="fig:silhouette416b"></span>
<img src="simpleSingleCell_files/figure-html/silhouette416b-1.png" alt="Barplot of silhouette widths for cells in each cluster. Each cluster is assigned a colour and cells with positive widths are coloured according to the colour of its assigned cluster. Any cell with a negative width is coloured according to the colour of the cluster that it is closest to. The average width for all cells in each cluster is shown, along with the average width for all cells in the dataset." width="672" />
<p class="caption">
Figure 12.2: Barplot of silhouette widths for cells in each cluster. Each cluster is assigned a colour and cells with positive widths are coloured according to the colour of its assigned cluster. Any cell with a negative width is coloured according to the colour of the cluster that it is closest to. The average width for all cells in each cluster is shown, along with the average width for all cells in the dataset.
</p>
</div>
<p>The silhouette width can be used to determine the parameter values that maximize the separation between clusters.
For example, we could vary the cut height or splitting depth in <code>cutreeDynamic</code> to maximize the average silhouette width across all cells.
This usually provides a satisfactory initial clustering for further examination.
However, keep in mind that the granularity of clustering is much like the magnification on a microscope.
Different views of the data can be obtained with different granularities, some of which may be suboptimal on measures of separation.
Users should not fixate on the clustering with the greatest separation if it does not provide the desired granularity for a particular biological question.</p>
<p>Most cells have relatively small silhouette positive widths in Figure <a href="clustering-cells-into-putative-subpopulations.html#fig:silhouette416b">12.2</a>, indicating that the separation between clusters is weak.
This may be symptomatic of over-clustering where clusters that are clearly defined on oncogene induction status are further split into subsets that are less well separated.
Nonetheless, we will proceed with the current clustering scheme in <code>my.clusters</code>, as it provides reasonable partitions for further characterization of heterogeneity.</p>
<p><strong>Comments from Aaron:</strong></p>
<ul>
<li>An alternative clustering strategy is to use a matrix of distances derived from correlations (e.g., as in <code>quickCluster</code>).
This is more robust to noise and normalization errors, but is also less sensitive to subtle changes in the expression profiles.</li>
<li>Both Ward’s criterion and complete linkage yield spherical, compact clusters.
In particular, complete linkage favours the formation of clusters with the same diameter.
This may be desirable in some cases but is less appropriate when subpopulations differ in their variance.
Thus, we typically use Ward’s criterion for our initial clustering.
Of course, it is simple (and recommended) to try other approaches provided that some assessment is performed, e.g., using the silhouette width.</li>
</ul>
</div>
<div id="detecting-marker-genes-between-clusters" class="section level2">
<h2><span class="header-section-number">12.2</span> Detecting marker genes between clusters</h2>
<p>Once putative subpopulations are identified by clustering, we can identify marker genes for each cluster using the <code>findMarkers</code> function.
This performs Welch <span class="math inline">\(t\)</span>-tests on the log-expression values for every gene and between every pair of clusters <span class="citation">(Soneson and Robinson <a href="#ref-soneson2018bias">2018</a>)</span>.
The aim is to test for DE in each cluster compared to the others while blocking on uninteresting factors such as the plate of origin.
The top DE genes are likely to be good candidate markers as they can effectively distinguish between cells in different clusters.</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb74-1" data-line-number="1">markers &lt;-<span class="st"> </span><span class="kw">findMarkers</span>(sce, my.clusters, <span class="dt">block=</span>sce<span class="op">$</span>Plate)</a></code></pre></div>
<p>For each cluster, the DE results of the relevant comparisons are consolidated into a single output table.
This allows a set of marker genes to be easily defined by taking the top DE genes from each pairwise comparison between clusters.
For example, to construct a marker set for cluster 1 from the top 10 genes of each comparison, one would filter <code>marker.set</code> to retain rows with <code>Top</code> less than or equal to 10.
Other statistics are also reported for each gene, including the adjusted p-values (see below) and the log-fold changes relative to every other cluster.</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb75-1" data-line-number="1">marker.set &lt;-<span class="st"> </span>markers[[<span class="st">&quot;1&quot;</span>]]</a>
<a class="sourceLine" id="cb75-2" data-line-number="2"><span class="kw">head</span>(marker.set, <span class="dv">10</span>)</a></code></pre></div>
<pre><code>## DataFrame with 10 rows and 6 columns
##             Top                  FDR           logFC.2           logFC.3
##       &lt;integer&gt;            &lt;numeric&gt;         &lt;numeric&gt;         &lt;numeric&gt;
## Aurkb         1 3.57731459036545e-32 -7.39137048940061 -6.77227082342766
## Tk1           1 6.14583238348605e-29 -4.79357274943543   -7.610521944668
## Cdca8         1 6.27977508120356e-23 -6.98353186925976 -5.14867209037853
## Myh11         1 7.46521687382308e-22  4.37714191593754  4.37854437192151
## Ccna2         2 1.80015331041228e-29  -7.4330496627687 -7.08153074893503
## Pclaf         2 2.31327100093205e-25 -5.56714921671886 -7.57990988274836
## Cks1b         2 2.06467827649085e-18 -6.81767090676048 -6.04001181928717
## Pirb          2 2.50560067866424e-15  5.19604859960812  5.24523582985688
## Ube2c         3 4.84313636309891e-28 -7.94593152928904 -4.23898926544224
## Rrm2          3 1.52352395791045e-24 -5.45829619585342 -7.75339990141175
##                 logFC.4             logFC.5
##               &lt;numeric&gt;           &lt;numeric&gt;
## Aurkb -1.95145075542623   -6.41735160332462
## Tk1   -3.52963403498811   -4.24172385898634
## Cdca8 -2.52622785978127   -7.26824410188525
## Myh11  4.45150905933443     0.9387643805397
## Ccna2 -2.55994736892407   -7.31965669537105
## Pclaf  -2.4323481751295    -5.2901842158707
## Cks1b -4.45638344358755   -6.40480503325219
## Pirb   5.86618518791929 -0.0110358172874627
## Ube2c -2.14802144694623   -6.67791590516897
## Rrm2  -3.18046561232799   -4.95579772214588</code></pre>
<p>We save the list of candidate marker genes for further examination.</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb77-1" data-line-number="1"><span class="kw">write.table</span>(marker.set, <span class="dt">file=</span><span class="st">&quot;416B_marker_1.tsv&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, </a>
<a class="sourceLine" id="cb77-2" data-line-number="2">    <span class="dt">quote=</span><span class="ot">FALSE</span>, <span class="dt">col.names=</span><span class="ot">NA</span>)</a></code></pre></div>
<p>We visualize the expression profiles of the top candidates to verify that the DE signature is robust (Figure <a href="clustering-cells-into-putative-subpopulations.html#fig:heatmapmarker416b">12.3</a>).
Most of the top markers have strong and consistent up- or downregulation in cells of cluster 1 compared to some or all of the other clusters.
A cursory examination of the heatmap indicates that cluster 1 contains oncogene-induced cells with strong downregulation of DNA replication and cell cycle genes.
This is consistent with the potential induction of senescence as an anti-tumorigenic response <span class="citation">(Wajapeyee et al. <a href="#ref-wajapeyee2010senescence">2010</a>)</span>.
A more comprehensive investigation of the function of these markers can be performed with gene set enrichment analyses, e.g., using <code>kegga</code> or <code>goana</code> from <em><a href="http://bioconductor.org/packages/limma">limma</a></em>.</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb78-1" data-line-number="1">top.markers &lt;-<span class="st"> </span><span class="kw">rownames</span>(marker.set)[marker.set<span class="op">$</span>Top <span class="op">&lt;=</span><span class="st"> </span><span class="dv">10</span>]</a>
<a class="sourceLine" id="cb78-2" data-line-number="2"><span class="kw">plotHeatmap</span>(sce, <span class="dt">features=</span>top.markers, <span class="dt">columns=</span><span class="kw">order</span>(sce<span class="op">$</span>cluster), </a>
<a class="sourceLine" id="cb78-3" data-line-number="3">    <span class="dt">colour_columns_by=</span><span class="kw">c</span>(<span class="st">&quot;cluster&quot;</span>, <span class="st">&quot;Plate&quot;</span>, <span class="st">&quot;Oncogene&quot;</span>),</a>
<a class="sourceLine" id="cb78-4" data-line-number="4">    <span class="dt">cluster_cols=</span><span class="ot">FALSE</span>, <span class="dt">center=</span><span class="ot">TRUE</span>, <span class="dt">symmetric=</span><span class="ot">TRUE</span>, <span class="dt">zlim=</span><span class="kw">c</span>(<span class="op">-</span><span class="dv">5</span>, <span class="dv">5</span>)) </a></code></pre></div>
<div class="figure"><span id="fig:heatmapmarker416b"></span>
<img src="simpleSingleCell_files/figure-html/heatmapmarker416b-1.png" alt="Heatmap of mean-centred and normalized log-expression values for the top set of markers for cluster 1 in the 416B dataset. Column colours represent the cluster to which each cell is assigned, the plate of origin or the oncogene induction status of each cell, as indicated by the legend." width="960" />
<p class="caption">
Figure 12.3: Heatmap of mean-centred and normalized log-expression values for the top set of markers for cluster 1 in the 416B dataset. Column colours represent the cluster to which each cell is assigned, the plate of origin or the oncogene induction status of each cell, as indicated by the legend.
</p>
</div>
<p>Many of the markers in Figure <a href="clustering-cells-into-putative-subpopulations.html#fig:heatmapmarker416b">12.3</a> are not uniquely up- or downregulated in the chosen cluster.
Testing for unique DE tends to be too stringent as it overlooks important genes that are expressed in two or more clusters.
For example, in a mixed population of CD4<sup>+</sup>-only, CD8<sup>+</sup>-only, double-positive and double-negative T cells, neither <em>Cd4</em> or <em>Cd8</em> would be detected as subpopulation-specific markers because each gene is expressed in two subpopulations.
With our approach, both of these genes will be picked up as candidate markers as they will be DE between at least one pair of subpopulations.
A combination of markers can then be chosen to characterize a subpopulation, which is more flexible than trying to find uniquely DE genes.</p>
<p>We strongly recommend selecting some markers for use in validation studies with an independent replicate population of cells.
The aim is to identify a corresponding subset of cells that express the upregulated markers and do not express the downregulated markers.
Ideally, a different technique for quantifying expression would also be used during validation, e.g., fluorescent <em>in situ</em> hybridisation or quantitative PCR.
This confirms that the subpopulation genuinely exists and is not an artifact of scRNA-seq or the computational analysis.</p>
<p><strong>Comments from Aaron:</strong></p>
<ul>
<li>By setting <code>direction=&quot;up&quot;</code>, <code>findMarkers</code> will only return genes that are upregulated in each cluster compared to the others.
This is convenient in highly heterogeneous populations to focus on genes that can immediately identify each cluster.
While lack of expression may also be informative, it is less useful for positive identification.</li>
<li>By setting <code>block=</code>, <code>findMarkers()</code> will perform pairwise tests between clusters using only cells on the same plate.
It will then combine <span class="math inline">\(p\)</span>-values from different plates using Stouffer’s Z method to obtain a single <span class="math inline">\(p\)</span>-value per gene.
An alternative approach is to use linear models via the <code>design=</code> argument - the differences between the two are discussed in the vignette.</li>
<li><code>findMarkers</code> can also be directed to find genes that are DE between the chosen cluster and <em>all</em> other clusters.
This should be done by setting <code>pval.type=&quot;all&quot;</code>, which defines the p-value for each gene as the maximum value across all pairwise comparisons involving the chosen cluster.
Combined with <code>direction=&quot;up&quot;</code>, this can be used to identify unique markers for each cluster.
However, this is sensitive to overclustering, as unique marker genes will no longer exist if a cluster is split into two smaller subclusters.</li>
<li>It must be stressed that the (adjusted) <em>p</em>-values computed here cannot be properly interpreted as measures of significance.
This is because the clusters have been empirically identified from the data.
<em><a href="http://bioconductor.org/packages/limma">limma</a></em> does not account for the uncertainty of clustering, which means that the <em>p</em>-values are much lower than they should be.
This is not a concern in other analyses where the groups are pre-defined.</li>
<li>The <code>overlapExprs</code> function may also be useful here, to prioritize candidates where there is clear separation between the distributions of expression values of different clusters.
This uses the Wilcoxon rank sum test to detect uneven mixing of the distributions of expression values between clusters.
By contrast, <code>findMarkers</code> uses t-tests and is primarily concerned with log-fold changes.</li>
</ul>
</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-langfelder2008defining">
<p>Langfelder, P., B. Zhang, and S. Horvath. 2008. “Defining clusters from a hierarchical cluster tree: the Dynamic Tree Cut package for R.” <em>Bioinformatics</em> 24 (5): 719–20.</p>
</div>
<div id="ref-soneson2018bias">
<p>Soneson, C., and M. D. Robinson. 2018. “Bias, robustness and scalability in single-cell differential expression analysis.” <em>Nat. Methods</em> 15 (4): 255–61.</p>
</div>
<div id="ref-wajapeyee2010senescence">
<p>Wajapeyee, N., S. Z. Wang, R. W. Serra, P. D. Solomon, A. Nagarajan, X. Zhu, and M. R. Green. 2010. “Senescence induction in human fibroblasts and hematopoietic progenitors by leukemogenic fusion proteins.” <em>Blood</em> 115 (24): 5057–60.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="visualizing-data-in-low-dimensional-space.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="concluding-remarks.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": ["simpleSingleCell.pdf", "simpleSingleCell.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
