<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Workflows for analyzing single-cell RNA-seq data with R/Bioconductor</title>
  <meta name="description" content="Workflows for analyzing single-cell RNA-seq data with R/Bioconductor">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="Workflows for analyzing single-cell RNA-seq data with R/Bioconductor" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Workflows for analyzing single-cell RNA-seq data with R/Bioconductor" />
  
  
  



<meta name="date" content="2018-10-25">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="overview.html">
<link rel="next" href="quality-control-on-the-cells.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Low Level scRNA-seq Workflow</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Intro</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#workflow-version-information"><i class="fa fa-check"></i><b>1.1</b> Workflow version information</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#motivation"><i class="fa fa-check"></i><b>1.2</b> Motivation</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#scrna-seq-data-analysis-with-bioconductor"><i class="fa fa-check"></i><b>1.3</b> scRNA-seq data analysis with Bioconductor</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#obtaining-a-count-matrix"><i class="fa fa-check"></i><b>1.4</b> Obtaining a count matrix</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#author-information"><i class="fa fa-check"></i><b>1.5</b> Author information</a><ul>
<li class="chapter" data-level="1.5.1" data-path="index.html"><a href="index.html#author-contributions"><i class="fa fa-check"></i><b>1.5.1</b> Author contributions</a></li>
<li class="chapter" data-level="1.5.2" data-path="index.html"><a href="index.html#competing-interests"><i class="fa fa-check"></i><b>1.5.2</b> Competing interests</a></li>
<li class="chapter" data-level="1.5.3" data-path="index.html"><a href="index.html#grant-information"><i class="fa fa-check"></i><b>1.5.3</b> Grant information</a></li>
<li class="chapter" data-level="1.5.4" data-path="index.html"><a href="index.html#acknowledgements"><i class="fa fa-check"></i><b>1.5.4</b> Acknowledgements</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="overview.html"><a href="overview.html"><i class="fa fa-check"></i><b>2</b> Overview</a></li>
<li class="chapter" data-level="3" data-path="setting-up-the-data.html"><a href="setting-up-the-data.html"><i class="fa fa-check"></i><b>3</b> Setting up the data</a><ul>
<li class="chapter" data-level="3.1" data-path="setting-up-the-data.html"><a href="setting-up-the-data.html#loading-in-the-count-matrix"><i class="fa fa-check"></i><b>3.1</b> Loading in the count matrix</a></li>
<li class="chapter" data-level="3.2" data-path="setting-up-the-data.html"><a href="setting-up-the-data.html#incorporating-cell-based-annotation"><i class="fa fa-check"></i><b>3.2</b> Incorporating cell-based annotation</a></li>
<li class="chapter" data-level="3.3" data-path="setting-up-the-data.html"><a href="setting-up-the-data.html#incorporating-gene-based-annotation"><i class="fa fa-check"></i><b>3.3</b> Incorporating gene-based annotation</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="quality-control-on-the-cells.html"><a href="quality-control-on-the-cells.html"><i class="fa fa-check"></i><b>4</b> Quality control on the cells</a><ul>
<li class="chapter" data-level="4.1" data-path="quality-control-on-the-cells.html"><a href="quality-control-on-the-cells.html#defining-the-quality-control-metrics"><i class="fa fa-check"></i><b>4.1</b> Defining the quality control metrics</a></li>
<li class="chapter" data-level="4.2" data-path="quality-control-on-the-cells.html"><a href="quality-control-on-the-cells.html#identifying-outliers-for-each-metric"><i class="fa fa-check"></i><b>4.2</b> Identifying outliers for each metric</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="classification-of-cell-cycle-phase.html"><a href="classification-of-cell-cycle-phase.html"><i class="fa fa-check"></i><b>5</b> Classification of cell cycle phase</a></li>
<li class="chapter" data-level="6" data-path="examining-gene-level-expression-metrics.html"><a href="examining-gene-level-expression-metrics.html"><i class="fa fa-check"></i><b>6</b> Examining gene-level expression metrics</a><ul>
<li class="chapter" data-level="6.1" data-path="examining-gene-level-expression-metrics.html"><a href="examining-gene-level-expression-metrics.html#inspecting-the-most-highly-expressed-genes"><i class="fa fa-check"></i><b>6.1</b> Inspecting the most highly expressed genes</a></li>
<li class="chapter" data-level="6.2" data-path="examining-gene-level-expression-metrics.html"><a href="examining-gene-level-expression-metrics.html#filtering-out-low-abundance-genes"><i class="fa fa-check"></i><b>6.2</b> Filtering out low-abundance genes</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="normalization-of-cell-specific-biases.html"><a href="normalization-of-cell-specific-biases.html"><i class="fa fa-check"></i><b>7</b> Normalization of cell-specific biases</a><ul>
<li class="chapter" data-level="7.1" data-path="normalization-of-cell-specific-biases.html"><a href="normalization-of-cell-specific-biases.html#using-the-deconvolution-method-to-deal-with-zero-counts"><i class="fa fa-check"></i><b>7.1</b> Using the deconvolution method to deal with zero counts</a></li>
<li class="chapter" data-level="7.2" data-path="normalization-of-cell-specific-biases.html"><a href="normalization-of-cell-specific-biases.html#computing-separate-size-factors-for-spike-in-transcripts"><i class="fa fa-check"></i><b>7.2</b> Computing separate size factors for spike-in transcripts</a></li>
<li class="chapter" data-level="7.3" data-path="normalization-of-cell-specific-biases.html"><a href="normalization-of-cell-specific-biases.html#applying-the-size-factors-to-normalize-gene-expression"><i class="fa fa-check"></i><b>7.3</b> Applying the size factors to normalize gene expression</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="modelling-the-technical-noise-in-gene-expression.html"><a href="modelling-the-technical-noise-in-gene-expression.html"><i class="fa fa-check"></i><b>8</b> Modelling the technical noise in gene expression</a></li>
<li class="chapter" data-level="9" data-path="removing-the-batch-effect.html"><a href="removing-the-batch-effect.html"><i class="fa fa-check"></i><b>9</b> Removing the batch effect</a></li>
<li class="chapter" data-level="10" data-path="denoising-expression-values-using-pca.html"><a href="denoising-expression-values-using-pca.html"><i class="fa fa-check"></i><b>10</b> Denoising expression values using PCA</a></li>
<li class="chapter" data-level="11" data-path="visualizing-data-in-low-dimensional-space.html"><a href="visualizing-data-in-low-dimensional-space.html"><i class="fa fa-check"></i><b>11</b> Visualizing data in low-dimensional space</a><ul>
<li class="chapter" data-level="11.1" data-path="visualizing-data-in-low-dimensional-space.html"><a href="visualizing-data-in-low-dimensional-space.html#with-pca"><i class="fa fa-check"></i><b>11.1</b> With PCA</a></li>
<li class="chapter" data-level="11.2" data-path="visualizing-data-in-low-dimensional-space.html"><a href="visualizing-data-in-low-dimensional-space.html#with-t-sne"><i class="fa fa-check"></i><b>11.2</b> With <em>t</em>-SNE</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="clustering-cells-into-putative-subpopulations.html"><a href="clustering-cells-into-putative-subpopulations.html"><i class="fa fa-check"></i><b>12</b> Clustering cells into putative subpopulations</a><ul>
<li class="chapter" data-level="12.1" data-path="clustering-cells-into-putative-subpopulations.html"><a href="clustering-cells-into-putative-subpopulations.html#defining-cell-clusters-from-expression-data"><i class="fa fa-check"></i><b>12.1</b> Defining cell clusters from expression data</a></li>
<li class="chapter" data-level="12.2" data-path="clustering-cells-into-putative-subpopulations.html"><a href="clustering-cells-into-putative-subpopulations.html#detecting-marker-genes-between-clusters"><i class="fa fa-check"></i><b>12.2</b> Detecting marker genes between clusters</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="concluding-remarks.html"><a href="concluding-remarks.html"><i class="fa fa-check"></i><b>13</b> Concluding remarks</a></li>
<li class="chapter" data-level="14" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>14</b> References</a></li>
<li class="chapter" data-level="" data-path="references-1.html"><a href="references-1.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/AJABADI/" target="blank">See on GitHub</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Workflows for analyzing single-cell RNA-seq data with R/Bioconductor</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="setting-up-the-data" class="section level1">
<h1><span class="header-section-number">Chapter 3</span> Setting up the data</h1>
<div id="loading-in-the-count-matrix" class="section level2">
<h2><span class="header-section-number">3.1</span> Loading in the count matrix</h2>
<p>Our first task is to load the count matrices into memory.
One matrix was generated for each plate of cells used in the study.
In each matrix, each row represents an endogenous gene or a spike-in transcript, and each column represents a cell.
Subsequently, the count in each entry of the matrix represents the number of reads mapped to a particular gene/transcript in a particular cell.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">plate1 &lt;-<span class="st"> </span><span class="kw">read.delim</span>(<span class="st">&quot;counts_Calero_20160113.tsv&quot;</span>, </a>
<a class="sourceLine" id="cb2-2" data-line-number="2">    <span class="dt">header=</span><span class="ot">TRUE</span>, <span class="dt">row.names=</span><span class="dv">1</span>, <span class="dt">check.names=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">plate2 &lt;-<span class="st"> </span><span class="kw">read.delim</span>(<span class="st">&quot;counts_Calero_20160325.tsv&quot;</span>, </a>
<a class="sourceLine" id="cb2-4" data-line-number="4">    <span class="dt">header=</span><span class="ot">TRUE</span>, <span class="dt">row.names=</span><span class="dv">1</span>, <span class="dt">check.names=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb2-5" data-line-number="5"></a>
<a class="sourceLine" id="cb2-6" data-line-number="6">gene.lengths &lt;-<span class="st"> </span>plate1<span class="op">$</span>Length <span class="co"># First column is the gene length.</span></a>
<a class="sourceLine" id="cb2-7" data-line-number="7">plate1 &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(plate1[,<span class="op">-</span><span class="dv">1</span>]) <span class="co"># Discarding gene length (as it is not a cell).</span></a>
<a class="sourceLine" id="cb2-8" data-line-number="8">plate2 &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(plate2[,<span class="op">-</span><span class="dv">1</span>])</a>
<a class="sourceLine" id="cb2-9" data-line-number="9"><span class="kw">rbind</span>(<span class="dt">Plate1=</span><span class="kw">dim</span>(plate1), <span class="dt">Plate2=</span><span class="kw">dim</span>(plate2))</a></code></pre></div>
<pre><code>##         [,1] [,2]
## Plate1 46703   96
## Plate2 46703   96</code></pre>
<p>We combine the two matrices into a single object for further processing.
This is done after verifying that the genes are in the same order between the two matrices.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw">stopifnot</span>(<span class="kw">identical</span>(<span class="kw">rownames</span>(plate1), <span class="kw">rownames</span>(plate2)))</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">all.counts &lt;-<span class="st"> </span><span class="kw">cbind</span>(plate1, plate2)</a></code></pre></div>
<p>For convenience, the count matrix is stored in a <code>SingleCellExperiment</code> object from the <em><a href="http://bioconductor.org/packages/SingleCellExperiment">SingleCellExperiment</a></em> package.
This allows different types of row- and column-level metadata to be stored alongside the counts for synchronized manipulation throughout the workflow.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw">library</span>(SingleCellExperiment)</a>
<a class="sourceLine" id="cb5-2" data-line-number="2">sce &lt;-<span class="st"> </span><span class="kw">SingleCellExperiment</span>(<span class="kw">list</span>(<span class="dt">counts=</span>all.counts))</a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="kw">rowData</span>(sce)<span class="op">$</span>GeneLength &lt;-<span class="st"> </span>gene.lengths</a>
<a class="sourceLine" id="cb5-4" data-line-number="4">sce</a></code></pre></div>
<pre><code>## class: SingleCellExperiment 
## dim: 46703 192 
## metadata(0):
## assays(1): counts
## rownames(46703): ENSMUSG00000102693 ENSMUSG00000064842 ... SIRV7
##   CBFB-MYH11-mcherry
## rowData names(1): GeneLength
## colnames(192): SLX-9555.N701_S502.C89V9ANXX.s_1.r_1
##   SLX-9555.N701_S503.C89V9ANXX.s_1.r_1 ...
##   SLX-11312.N712_S508.H5H5YBBXX.s_8.r_1
##   SLX-11312.N712_S517.H5H5YBBXX.s_8.r_1
## colData names(0):
## reducedDimNames(0):
## spikeNames(0):</code></pre>
<p>We identify the rows corresponding to ERCC spike-in transcripts from the row names.
We store this information in the <code>SingleCellExperiment</code> object for future use.
This is necessary as spike-ins require special treatment in downstream steps such as normalization.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw">isSpike</span>(sce, <span class="st">&quot;ERCC&quot;</span>) &lt;-<span class="st"> </span><span class="kw">grepl</span>(<span class="st">&quot;^ERCC&quot;</span>, <span class="kw">rownames</span>(sce))</a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="kw">summary</span>(<span class="kw">isSpike</span>(sce, <span class="st">&quot;ERCC&quot;</span>))</a></code></pre></div>
<pre><code>##    Mode   FALSE    TRUE 
## logical   46611      92</code></pre>
<p>This dataset is slightly unusual in that it contains information from another set of spike-in transcripts, the Spike-In RNA Variants (SIRV) set.
For simplicity, we will only use the ERCC spike-ins in this analysis.
Thus, we must remove the rows corresponding to the SIRV transcripts prior to further analysis, which can be done simply by subsetting the <code>SingleCellExperiment</code> object.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">is.sirv &lt;-<span class="st"> </span><span class="kw">grepl</span>(<span class="st">&quot;^SIRV&quot;</span>, <span class="kw">rownames</span>(sce))</a>
<a class="sourceLine" id="cb9-2" data-line-number="2">sce &lt;-<span class="st"> </span>sce[<span class="op">!</span>is.sirv,] </a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="kw">summary</span>(is.sirv)</a></code></pre></div>
<pre><code>##    Mode   FALSE    TRUE 
## logical   46696       7</code></pre>
<p><strong>Comments from Aaron:</strong></p>
<ul>
<li>Some feature-counting tools will report mapping statistics in the count matrix (e.g., the number of unaligned or unassigned reads).
While these values can be useful for quality control, they would be misleading if treated as gene expression values.
Thus, they should be removed (or at least moved to the <code>colData</code>) prior to further analyses.</li>
<li>Be aware of using the <code>^ERCC</code> regular expression for human data where the row names of the count matrix are gene symbols.
An ERCC gene family actually exists in human annotation, so this would result in incorrect identification of genes as spike-in transcripts.
This problem can be avoided by publishing count matrices with standard identifiers (e.g., Ensembl, Entrez).</li>
</ul>
</div>
<div id="incorporating-cell-based-annotation" class="section level2">
<h2><span class="header-section-number">3.2</span> Incorporating cell-based annotation</h2>
<p>We load in the metadata for each library/cell from the <code>sdrf.txt</code> file.
It is important to check that the rows of the metadata table are in the same order as the columns of the count matrix.
Otherwise, incorrect metadata will be assigned to each cell.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">metadata &lt;-<span class="st"> </span><span class="kw">read.delim</span>(lun.sdrf, <span class="dt">check.names=</span><span class="ot">FALSE</span>, <span class="dt">header=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb11-2" data-line-number="2">m &lt;-<span class="st"> </span><span class="kw">match</span>(<span class="kw">colnames</span>(sce), metadata[[<span class="st">&quot;Source Name&quot;</span>]]) <span class="co"># Enforcing identical order.</span></a>
<a class="sourceLine" id="cb11-3" data-line-number="3"><span class="kw">stopifnot</span>(<span class="kw">all</span>(<span class="op">!</span><span class="kw">is.na</span>(m))) <span class="co"># Checking that nothing&#39;s missing.</span></a>
<a class="sourceLine" id="cb11-4" data-line-number="4">metadata &lt;-<span class="st"> </span>metadata[m,]</a>
<a class="sourceLine" id="cb11-5" data-line-number="5"><span class="kw">head</span>(<span class="kw">colnames</span>(metadata))</a></code></pre></div>
<pre><code>## [1] &quot;Source Name&quot;                &quot;Comment[ENA_SAMPLE]&quot;       
## [3] &quot;Comment[BioSD_SAMPLE]&quot;      &quot;Characteristics[organism]&quot; 
## [5] &quot;Characteristics[cell line]&quot; &quot;Characteristics[cell type]&quot;</code></pre>
<p>We only retain relevant metadata fields to avoid storing unnecessary information in the <code>colData</code> of the <code>SingleCellExperiment</code> object.
In particular, we keep the plate of origin (i.e., <code>block</code>) and phenotype of each cell.
The second field is relevant as all of the cells contain a <em>CBFB-MYH11</em> oncogene, but the expression of this oncogene is only induced in a subset of the cells.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw">colData</span>(sce)<span class="op">$</span>Plate &lt;-<span class="st"> </span><span class="kw">factor</span>(metadata[[<span class="st">&quot;Factor Value[block]&quot;</span>]])</a>
<a class="sourceLine" id="cb13-2" data-line-number="2">pheno &lt;-<span class="st"> </span>metadata[[<span class="st">&quot;Factor Value[phenotype]&quot;</span>]]</a>
<a class="sourceLine" id="cb13-3" data-line-number="3"><span class="kw">levels</span>(pheno) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;induced&quot;</span>, <span class="st">&quot;control&quot;</span>)</a>
<a class="sourceLine" id="cb13-4" data-line-number="4"><span class="kw">colData</span>(sce)<span class="op">$</span>Oncogene &lt;-<span class="st"> </span>pheno</a>
<a class="sourceLine" id="cb13-5" data-line-number="5"><span class="kw">table</span>(<span class="kw">colData</span>(sce)<span class="op">$</span>Oncogene, <span class="kw">colData</span>(sce)<span class="op">$</span>Plate)</a></code></pre></div>
<pre><code>##          
##           20160113 20160325
##   induced       48       48
##   control       48       48</code></pre>
</div>
<div id="incorporating-gene-based-annotation" class="section level2">
<h2><span class="header-section-number">3.3</span> Incorporating gene-based annotation</h2>
<p>Feature-counting tools typically report genes in terms of standard identifiers from Ensembl or Entrez.
These identifiers are used as they are unambiguous and highly stable.
However, they are difficult to interpret compared to the gene symbols which are more commonly used in the literature.
Given the Ensembl identifiers, we obtain the corresponding gene symbols using annotation packages like <em><a href="http://bioconductor.org/packages/org.Mm.eg.db">org.Mm.eg.db</a></em>.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="kw">library</span>(org.Mm.eg.db)</a>
<a class="sourceLine" id="cb15-2" data-line-number="2">symb &lt;-<span class="st"> </span><span class="kw">mapIds</span>(org.Mm.eg.db, <span class="dt">keys=</span><span class="kw">rownames</span>(sce), <span class="dt">keytype=</span><span class="st">&quot;ENSEMBL&quot;</span>, <span class="dt">column=</span><span class="st">&quot;SYMBOL&quot;</span>)</a>
<a class="sourceLine" id="cb15-3" data-line-number="3"><span class="kw">rowData</span>(sce)<span class="op">$</span>ENSEMBL &lt;-<span class="st"> </span><span class="kw">rownames</span>(sce)</a>
<a class="sourceLine" id="cb15-4" data-line-number="4"><span class="kw">rowData</span>(sce)<span class="op">$</span>SYMBOL &lt;-<span class="st"> </span>symb</a>
<a class="sourceLine" id="cb15-5" data-line-number="5"><span class="kw">head</span>(<span class="kw">rowData</span>(sce))</a></code></pre></div>
<pre><code>## DataFrame with 6 rows and 3 columns
##   GeneLength            ENSEMBL      SYMBOL
##    &lt;integer&gt;        &lt;character&gt; &lt;character&gt;
## 1       1070 ENSMUSG00000102693          NA
## 2        110 ENSMUSG00000064842          NA
## 3       6094 ENSMUSG00000051951        Xkr4
## 4        480 ENSMUSG00000102851          NA
## 5       2819 ENSMUSG00000103377          NA
## 6       2233 ENSMUSG00000104017          NA</code></pre>
<p>It is often desirable to rename the row names of <code>sce</code> to the gene symbols, as these are easier to interpret.
However, this requires some work to account for missing and duplicate symbols.
The code below will replace missing symbols with the Ensembl identifier and concatenate duplicated symbols with the (unique) Ensembl identifiers.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1">new.names &lt;-<span class="st"> </span><span class="kw">rowData</span>(sce)<span class="op">$</span>SYMBOL</a>
<a class="sourceLine" id="cb17-2" data-line-number="2">missing.name &lt;-<span class="st"> </span><span class="kw">is.na</span>(new.names)</a>
<a class="sourceLine" id="cb17-3" data-line-number="3">new.names[missing.name] &lt;-<span class="st"> </span><span class="kw">rowData</span>(sce)<span class="op">$</span>ENSEMBL[missing.name]</a>
<a class="sourceLine" id="cb17-4" data-line-number="4">dup.name &lt;-<span class="st"> </span>new.names <span class="op">%in%</span><span class="st"> </span>new.names[<span class="kw">duplicated</span>(new.names)]</a>
<a class="sourceLine" id="cb17-5" data-line-number="5">new.names[dup.name] &lt;-<span class="st"> </span><span class="kw">paste0</span>(new.names, <span class="st">&quot;_&quot;</span>, <span class="kw">rowData</span>(sce)<span class="op">$</span>ENSEMBL)[dup.name]</a>
<a class="sourceLine" id="cb17-6" data-line-number="6"><span class="kw">rownames</span>(sce) &lt;-<span class="st"> </span>new.names</a>
<a class="sourceLine" id="cb17-7" data-line-number="7"><span class="kw">head</span>(<span class="kw">rownames</span>(sce))</a></code></pre></div>
<pre><code>## [1] &quot;ENSMUSG00000102693&quot; &quot;ENSMUSG00000064842&quot; &quot;Xkr4&quot;              
## [4] &quot;ENSMUSG00000102851&quot; &quot;ENSMUSG00000103377&quot; &quot;ENSMUSG00000104017&quot;</code></pre>
<p>We also determine the chromosomal location for each gene using the <em><a href="http://bioconductor.org/packages/TxDb.Mmusculus.UCSC.mm10.ensGene">TxDb.Mmusculus.UCSC.mm10.ensGene</a></em> package.
This will be useful later as several quality control metrics will be computed from rows corresponding to mitochondrial genes.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="kw">library</span>(TxDb.Mmusculus.UCSC.mm10.ensGene)</a>
<a class="sourceLine" id="cb19-2" data-line-number="2">location &lt;-<span class="st"> </span><span class="kw">mapIds</span>(TxDb.Mmusculus.UCSC.mm10.ensGene, <span class="dt">keys=</span><span class="kw">rowData</span>(sce)<span class="op">$</span>ENSEMBL, </a>
<a class="sourceLine" id="cb19-3" data-line-number="3">    <span class="dt">column=</span><span class="st">&quot;CDSCHROM&quot;</span>, <span class="dt">keytype=</span><span class="st">&quot;GENEID&quot;</span>)</a>
<a class="sourceLine" id="cb19-4" data-line-number="4"><span class="kw">rowData</span>(sce)<span class="op">$</span>CHR &lt;-<span class="st"> </span>location</a>
<a class="sourceLine" id="cb19-5" data-line-number="5"><span class="kw">summary</span>(location<span class="op">==</span><span class="st">&quot;chrM&quot;</span>)</a></code></pre></div>
<pre><code>##    Mode   FALSE    TRUE    NA&#39;s 
## logical   22428      13   24255</code></pre>
<p>Alternatively, annotation from BioMart resources can be directly added to the object using the <code>getBMFeatureAnnos</code> function from <em><a href="http://bioconductor.org/packages/scater">scater</a></em>.
This may be more convenient than the approach shown above, but depends on an available internet connection to the BioMart databases.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="overview.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="quality-control-on-the-cells.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": ["simpleSingleCell.pdf", "simpleSingleCell.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
